src/types/bytes.rs

```diff
@@ -11,6 +11,29 @@ use std::str;
 /// Represents a Python `bytes` object.
 ///
 /// This type is immutable.
+///
+/// # Equality
+///
+/// For convenience, [`Bound<'py, PyBytes>`] implements [`PartialEq<str>`] to allow comparing the
+/// data in the Python bytes to a Rust `[u8]`.
+///
+/// This is not always the most appropriate way to compare Python bytes, as Python bytes subclasses
+/// may have different equality semantics. In situations where subclasses overriding equality might be
+/// relevant, use [`PyAnyMethods::eq`], at cost of the additional overhead of a Python method call.
+///
+/// ```rust
+/// # use pyo3::prelude::*;
+/// use pyo3::types::PyBytes;
+///
+/// # Python::with_gil(|py| {
+/// let py_bytes = PyBytes::new_bound(py, b"foo");
+/// // via PartialEq<[u8]>
+/// assert_eq!(py_bytes, b"foo");
+///
+/// // via Python equality
+/// assert!(py_bytes.as_any().eq(b"foo").unwrap());
+/// # });
+/// ```
```

Comment:
I assume that we want similar documentation as we have for PyString. I copied the comments from PyString here and made a few changes. LMK if there are any details I missed here.

Reply:
Looks good to me.